contracts/dao-dao-core/src/contract.rs

```diff
@@ -159,8 +167,10 @@ pub fn execute_pause(
     sender: Addr,
     pause_duration: Duration,
 ) -> Result<Response, ContractError> {
-    // Only the core contract may call this method.
-    if sender != env.contract.address {
+    let admin = ADMIN.load(deps.storage)?;
+
+    // Only the core contract or admin may call this method.
+    if sender != env.contract.address && sender != admin {
```

Comment:
hmm, maybe only the admin should call this message. are there other messages that allow both the core contract and the admin?

Reply:
Ok I see the pattern is usually just checking for the sender being the dao-core, and the admin will execute through the `ExecuteAdminMsgs`.
I'll keep that same pattern and also allow the ExecuteAdminMsgs through when paused.

Reply:
actually, i think it makes sense that *only* the admin can call unpause, because ofc the SubDAO can't unpause itself normally.

hm that brings up another question actually:

should the parent be able act as the SubDAO if the SubDAO is paused?

this feels like an arbitrary design decision i don't know how to make. @bekauz @JakeHartnell any thoughts?

Reply:
i sent a message in the discord to solicit some community input:
https://discord.com/channels/895922260047720449/895926354728603659/1202665603446603847

Reply:
Actually, Unpause should only be called directly by the admin. A dao can unpause itself if admin, and only a parent dao should be able to unpause the subDao.

Reply:
right well a DAO can't unpause itself ever, because if it's paused, it can't execute messages. that's why i think it's safe to just check the admin.

Reply:
ok, i think a parent DAO may want to execute messages on behalf of a SubDAO when it's paused for security purposes. maybe the SubDAO is being taken over and the DAO needs to rescue funds or update membership.

Reply:
We'd have to refactor the way authorization works. Right now it's Admin -> ExecuteAdminMsgs, which calls DAO methods from DAO, because the actions are only checking to see if the sender is the dao itself. The msg loses some context.

Maybe we get rid of ExecuteAdminMsgs and create a helper method for checking authorization at each call based off admin, paused, and dao?

Reply:
why would we have to refactor that? what you said earlier about allowing `unpause` and `execute_admin_msgs` through while paused should be fine, right? as long as both `unpause` and `execute_admin_msgs` just check that sender is the admin.

Reply:
Because `execute_admin_msgs` is just passing the msgs through to be executed as the DAO itself using `.add_message(msgs)`. 

Take for example a paused DAO where we allow admin messages:
Admin messages pass through just fine.
The msgs are now attempting to be executed with `info.sender` being `env.contract.address`, but the message will now not be of type execute_admin or unpause and fail mesage check at the beginning.

Reply:
ah i see the confusion. `.add_message(msgs)` is the actual chain-level "i want to execute this message as myself" in a contract. those do not get routed through execute (unless of course the parent DAO tries to tell the SubDAO to execute a message on itself, such as updating core info). so i'm 99% sure that allowing both `unpause` and `execute_admin_msgs` will do what we want.

that actually brings up another use case: what if the admin wants to execute messages on the SubDAO while it's paused?

now i'm thinking that when a SubDAO is paused, the admin should be allowed to send any message it would normally be allowed to send (i.e. entirely skip the paused check if the admin is the sender), and any other sender errors out when paused. i believe this is safe because it will still prevent proposal modules from executing anything, and it will allow the admin to execute anything they want, including unpause.

Reply:
@bekauz pls sense-check everything i just said ^

Reply:
first, i'm a little paranoid about how this can potentially impact the veto flow. could we get some tests done to cover pausing/unpausing dynamics with a veto committee subdao?

same goes for the `execute_admin_msgs` discussion. we should have tests to cover those cases, because even though this is a straightforward pr, it impacts other flows.

Reply:
Ahh ok, I was confused about the admin telling the subdao to execute on itself, and it makes sense that we're really trying to lock down the proposal modules from further executing.

Reply:
Would it make sense to only check the paused state in the `execute_proposal_hook` method?

Reply:
imho that would be a little more confusing for someone not familiar with our repo, wdyt?